src/ui/cmake/Dependencies.cmake

```diff
@@ -45,15 +45,21 @@ FetchContent_Declare(
         GIT_TAG main # todo: use proper release once available
 )
 
+FetchContent_Declare(
+    function2
+    GIT_REPOSITORY https://github.com/Naios/function2.git
+    GIT_TAG 4.2.2
+)
```

Comment:
This looks like a lightweight enough dependency (1,8k lines header) but since we are trying to keep the number of required dependencies low, i think it would be good to just have 2 sentences of reasoning why this is needed.